src/util/fields.py

```diff
@@ -884,14 +905,39 @@ def _convert_field_to_property(
 
 
 def convert_binary_field_to_attachment(cr, model, field, encoded=True, name_field=None):
+    _extract_data_as_attachment(cr, model, field, encoded, name_field)
+    # free PG space
+    table = table_of_model(cr, model)
+    remove_column(cr, table, field)
+
+
+def _extract_data_as_attachment(cr, model, field, encoded=True, name_field=None, set_res_field=True):
     _validate_model(model)
     table = table_of_model(cr, model)
     if not column_exists(cr, table, field):
         return
-    name_query = "COALESCE({0}, '{1}('|| id || ').{2}')".format(
-        name_field if name_field else "NULL",
-        model.title().replace(".", ""),
-        field,
+    name_query = cr.mogrify(
+        format_query(cr, "COALESCE({}, CONCAT(%s, '(', id,  ').', %s)", name_field if name_field else SQLStr("NULL")),
+        [model.title().replace(".", ""), field],
+    ).decode()
+
+    if not column_exists(cr, "ir_attachment", "res_field"):
```

Comment:
Hm... was it a mistake in the previous implementation? We always assumed res_field was there.

Reply:
No. This function was explicitly written for handling the storage of the binary fields (`bytea` to `attachment`), and it's not possible without the `res_field` column.

Now that we repurpose this function, we need to handle the case.